### PR TITLE
Don't parse out a local declaration where there clearly isn't one.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
If we see:

```c#
  Identifier1
  Identifier2();
```

Don't parse that out as a declaration of a local called "Identifier2".

Fixes: https://github.com/dotnet/roslyn/issues/4014

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler  and @dotnet/roslyn-ide

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs [hunk 1]
```diff
@@ -4709,6 +4709,7 @@ private VariableDeclaratorSyntax ParseVariableDeclarator(TypeSyntax parentType,
                                 SyntaxFacts.IsBinaryExpressionOperatorToken(currentTokenKind);
 
                             if (currentTokenKind == SyntaxKind.DotToken ||
+                                currentTokenKind == SyntaxKind.OpenParenToken ||
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: Please update the comment above that explains this code.

**Comment by gafter** [comment]
Other than that LGTM.

File: src/Compilers/CSharp/Portable/Parser/LanguageParser.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/LanguageParser.cs]: Will do!

**Comment by jasonmalinowski** [comment]
Late but :+1:.

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**